### PR TITLE
(QENG-5302) add 'packaging_platform' values

### PR DESCRIPTION
**Pull request by kevpl**
more maps!

This change adds new 'packaging_platform' values for the OSes
we build packages for. This will allow beaker users to use the
new install methods created in BKR-1118 easier.

[skip ci]

**Note** that this work isn't complete at this point & will not pass testing.
This is because the generated test fixtures haven't been changed yet.
I haven't changed them because it would make this PR much harder
to deal with. I'd like to get this reviewed & ready to go, then add
another commit for fixture generation that can be reviewed separately
before merge.

# TODOs Before Merge

- get sign-off from these teams:
  - [x] RE
  - ~~Windows~~ (RE gave confidence that we don't need windows team approval specifically)
- [ ] once that's done, create the generated test fixture updates

### CODE REVIEW

**Comment by kevpl** [comment]
@melissa here's the work we talked about. I created it by looking at the YAML file generated for the sha `c6a76afc1522c139721626452cd6eda8d7815817` of puppet-agent.

I've added a log of my work auditing all of the platforms in the data map in beaker-hostgenerator & comparing them to the `platform_data` values from that YAML file in [this gist](https://gist.github.com/kevpl/d18cc39576d0bef72982ec976a09c224). Would you take a look at that log & verify that the values I added & I didn't add entries for are correct?

**Comment by melissa** [comment]
We're actually only building two packages for windows, the x64 msi and the x86 msi. Those can be installed on all different versions of windows as long as the architecture is consistent. 2012 is just the version we build on, and shouldn't matter for testing purposes.

**Comment by melissa** [comment]
I noticed a few entries in `data.rb` that I would expect to have `packaging_platform` set do not have it. AIX, for example, doesn't have that, but we definitely support AIX in the packaging repo. I haven't gone through it entry by entry yet, but I will

All of the `packaging_platform` entries we support currently are defined in https://github.com/puppetlabs/packaging/blob/master/lib/packaging/platforms.rb. If you want to verify the format different entries would take, you can check that. For example:
```
      'aix' => {
        '5.3' => { :architectures => ['power'], :repo => false, :package_format => 'rpm', },
        '6.1' => { :architectures => ['power'], :repo => false, :package_format => 'rpm', },
        '7.1' => { :architectures => ['power'], :repo => false, :package_format => 'rpm', },
      },
```
would give me
`aix-5.3-power`
`aix-6.1-power`
`aix-7.1-power`

and 
```
      'sles' => {
        '11' => { :architectures => ['i386', 'x86_64', 's390x'], :repo => true, :package_format => 'rpm', :signature_format => 'v3', },
        '12' => { :architectures => ['x86_64', 's390x', 'ppc64le'], :repo => true, :package_format => 'rpm', :signature_format => 'v4', },
      },
```
would give me
`sles-11-i386`
`sles-11-x86_64`
`sles-11-s390x`
`sles-12-x86_64`
`sles-12-s390x`
`sles-12-ppc64le`

File: lib/beaker-hostgenerator/data.rb [hunk 1]
```diff
@@ -939,7 +995,8 @@ def osinfo
         },
         'windows2012-64' => {
           :general => {
-            'platform' => 'windows-2012-64',
+            'platform'           => 'windows-2012-64',
+            'packaging_platform' => 'windows-2012-x64',
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: Every windows entry can be updated to include `'packaging_platform' => 'windows-2012-x64',` or `'packaging_platform' => 'windows-2012-x84',`. The `2012` is arbitrary. We just need a value there for the automation. In reality, the windows msi are not version specific.

**Comment by melissa** [comment]
entries that are missing:

- aix (7.2 == 7.1 for our purposes)
- fedora 26 (although we have yet to add this to the packaging platform_info hash, it will be added at some point. Any reason not to just add it now so we don't have to go back and add it later?)
- huaweios = `huaweios-6-powerpc`, even though we're dropping support for huaweios in the puppet5 series. I'll let you decide if you want to add it in or not
- `opensuse` would just be `sles`, right?

File: lib/beaker-hostgenerator/data.rb [hunk 2]
```diff
@@ -502,97 +529,110 @@ def osinfo
         },
         'redhat5-32' => {
           :general => {
-            'platform' => 'el-5-i386'
+            'platform'          => 'el-5-i386',
+            'packging_platform' => 'el-5-i386'
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: typo

File: lib/beaker-hostgenerator/data.rb [hunk 3]
```diff
@@ -502,97 +529,110 @@ def osinfo
...
           },
           :vmpooler => {
             'template' => 'redhat-5-i386'
           }
         },
         'redhat5-64' => {
           :general => {
-            'platform' => 'el-5-x86_64'
+            'platform'          => 'el-5-x86_64',
+            'packging_platform' => 'el-5-x86_64'
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: typo

File: lib/beaker-hostgenerator/data.rb [hunk 4]
```diff
@@ -502,97 +529,110 @@ def osinfo
...
           },
           :vmpooler => {
             'template' => 'redhat-5-x86_64'
           }
         },
         'redhat6-32' => {
           :general => {
-            'platform' => 'el-6-i386'
+            'platform'          => 'el-6-i386',
+            'packging_platform' => 'el-6-i386'
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: typo

File: lib/beaker-hostgenerator/data.rb [hunk 5]
```diff
@@ -502,97 +529,110 @@ def osinfo
...
           },
           :vmpooler => {
             'template' => 'redhat-6-x86_64'
           }
         },
         'redhat6-S390X' => {
           :general => {
-            'platform' => 'el-6-s390x'
+            'platform'          => 'el-6-s390x',
+            'packging_platform' => 'el-6-s390x'
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: typo

File: lib/beaker-hostgenerator/data.rb [hunk 6]
```diff
@@ -502,97 +529,110 @@ def osinfo
...
           },
           :vmpooler => {
             'template' => 'scientific-5-x86_64'
           }
         },
         'scientific6-32' => {
           :general => {
-            'platform' => 'el-6-i386'
+            'platform'          => 'el-6-i386',
+            'packging_platform' => 'el-6-i386'
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: typo

File: lib/beaker-hostgenerator/data.rb [hunk 7]
```diff
@@ -502,97 +529,110 @@ def osinfo
...
           },
           :vmpooler => {
             'template' => 'scientific-6-x86_64'
           }
         },
         'scientific7-64' => {
           :general => {
-            'platform' => 'el-7-x86_64'
+            'platform'          => 'el-7-x86_64',
+            'packging_platform' => 'el-7-x86_64'
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: typo

File: lib/beaker-hostgenerator/data.rb [hunk 8]
```diff
@@ -182,15 +189,17 @@ def osinfo
         },
         'ciscoxr-64' => {
           :general => {
-            'platform' => 'cisco_ios_xr-6-x86_64'
+            'platform'           => 'cisco_ios_xr-6-x86_64',
+            'packaging_platform' => 'cisco_wrlinux-7-x86_64'
```
Comment (melissa) [inline lib/beaker-hostgenerator/data.rb]: this whole thing for cisco is terrifying...

**Comment by kevpl** [comment]
@melissa I think I've got all the points you made corrected. Let me know if I've missed any or gotten any wrong.

File: lib/beaker-hostgenerator/data.rb [hunk 9]
```diff
@@ -398,55 +422,62 @@ def osinfo
         },
         'opensuse11-32' => {
```
Comment (shrug) [inline lib/beaker-hostgenerator/data.rb]: We don't do anything at all with opensuse do we?

**Comment by shrug** [comment]
One question about opensuse generally, but otherwise looks good to me

File: lib/beaker-hostgenerator/data.rb (hunk 9 above)
Comment (kevpl) [reply lib/beaker-hostgenerator/data.rb]: @shrug who should we direct this question towards? I assume someone like @joshcooper would be able to answer it, but if he doesn't have time for this, who would be a good person to ping to get this information?

Comment (melissa) [reply lib/beaker-hostgenerator/data.rb]: In order to move forward with this, let's just remove `packaging_platform` from the `opensuse` test targets. It sounds like I was wrong in my assumption the sles packages would just work for opensuse. If this becomes an issue in the future, we can figure out a better solution.

**Comment by kevpl** [comment]
@shrug @melissa removed the `packaging_platform` values from the opensuse sections. Let me know if there's anything else you're seeing here.

**Comment by melissa** [comment]
@kevpl should we get input on this from any other teams/stakeholders before it gets merged?

**Comment by kevpl** [comment]
now that the _real_ work has been approved by RE, the last thing to do before final review & merge is for me to update the testing fixtures, so this PR is now waiting on me. Hopefully I'll finish that by end of day today.

**Comment by kevpl** [comment]
@nwolfe I've re-generated test fixtures in 2ddd304, so now this PR is truly out-of-hand like all the others that affect the fixtures. If you click on the first commit (3f9088c), you can see the real changes that were approved by @melissa & @shrug, and that the new commit is just the test fixtures re-generated.

Let me know if I've done this wrong or misunderstood any of the stuff we were talking about for this PR, and hopefully we can get this through soon. Thanks for your help!

**Comment by melissa** [comment]
Oh god, this turned truly terrifying with the regenerated fixtures.....

@nwolfe if we could get this in soon, that would be super fantastic!

**Pull request merged**